### ISSUE TITLE
Some tests broke due to a change in UnitTest.wait

### ISSUE THREAD

**Issue by patrickdupuis** [supercollider/supercollider#5366]
## Environment

* SuperCollider version: current `develop`

## Issue

A handful of UnitTests broke when `UnitTest.wait` changed in https://github.com/supercollider/supercollider/commit/2aa3374e1e52b7db3f7708498810dc6032a1d082

This breakage is pretty easy to spot. You'll see an error like:
```
^^ ERROR: Message 'test' not understood.
Perhaps you misspelled 'dist', or meant to call 'test' on another receiver?
```

These tests need to be modified so that calls to `UnitTest.wait` are passed a Condition instead of a Function or Boolean.

- TestFilterUGens
- TestNode_Server
- TestServer_boot
- TestMixedBundleTester
- TestCoreUGens
- TestDocument

See discussions in #4659 and #5363
Related issue #5359

**Comment by brianlheim** [comment]
Hey @patrickdupuis, here's what I propose we do. I've opened #5394 which would allow us to proceed in the direction @jamshark70 wanted in #5368. I think we'll have to do that work eventually, so I'd like to not change things twice. But, I also recognize this is a time sensitive problem. How about if #5394 has not been reviewed and merged within a week or so, we reopen #5363 and try to get that merged as a temporary workaround?

**Comment by patrickdupuis** [comment]
I'm ok with waiting on #5394 for now. Let's see how things go this week.

**Comment by brianlheim** [comment]
@patrickdupuis sorry I didn't revisit this this weekend. I wanted to check in and see how you are feeling about this now. I think there is good momentum on the CondVar PR and it may be merged soon, but I also know that's pure optimism on my part. Whatever you want to do next is good with me.

**Comment by patrickdupuis** [comment]
Looks like #5394 will be merged soon. We can fix the broken tests in a follow-up PR. This can stay closed.

**Comment by patrickdupuis** [comment]
oops closed by mistake.

**Comment by jparmentervf** [comment]
@patrickdupuis - how are things looking for this at the moment?
